src/atc/configurator/_cli/generate_keys_file.py

```
@@ -0,0 +1,62 @@
+import argparse
+import os.path
+from io import StringIO
+
+from atc.exceptions.cli_exceptions import AtcCliCheckFailed
+
+
+def setup_parser(parser: argparse.ArgumentParser):
+    parser.set_defaults(func=generate_keys_file)
+    parser.add_argument("-o", "--output_file", type=str, default="")
+    parser.add_argument(
+        "-c",
+        "--check-only",
+        dest="check",
+        action="store_true",
+        help="Only check, don't update the output file",
+    )
+    parser.set_defaults(check=False)
+
+
+def generate_keys_file(self, options):
+
+    writer = StringIO()
+    writer.write("# AUTO GENERATED FILE\n# contains all atc.Configurator keys\n\n")
+
+    for key in self.all_keys():
+        writer.write(f"{key} = {repr(key)}\n")
+
+    writer.seek(0)
+    new_conts = writer.read()
+
+    try:
+        import black
+
+        new_conts = black.format_file_contents(
+            new_conts, fast=False, mode=black.FileMode()
+        )
+    except ModuleNotFoundError:
+        pass
+
+    if not options.output_file:
+        print(new_conts, end="")
+        return 0
+    else:
+        if options.check:
+            if os.path.exists(options.output_file):
```

fail fast pattern instead